python_modules/dagster/dagster/_core/execution/job_backfill.py

```diff
@@ -185,7 +187,29 @@ def _get_partitions_chunk(
     backfill_runs = instance.get_runs(
         RunsFilter(tags=DagsterRun.tags_for_backfill_id(backfill_job.backfill_id))
     )
-    completed_partitions = set([run.tags.get(PARTITION_NAME_TAG) for run in backfill_runs])
+    if backfill_policy and backfill_policy.max_partitions_per_run != 1:
```

Comment:
it looks like the corresponding logic in asset backfills doesn't gate specifically on the backfill policy and just looks at the tags: https://github.com/dagster-io/dagster/blob/4029c835afa75d18a4ce96782e0e706f494efb43/python_modules/dagster/dagster/_core/execution/asset_backfill.py#L1707-L1729 - you could consider mimicing that

Reply:
yeah i'll switch it, that's a safer approach